File: typescript/base-hmi-types/src/screens/controls/HmiDataGridColumn.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base/HmiProperty.js";
import { HmiDataGridColumnWidthMode } from "./HmiDataGridColumnWidthMode.js";
import { HmiDataGridSortDirection } from "./HmiDataGridSortDirection.js";

export class HmiDataGridColumn {
  name?: string;
  dataType?: string;
  visible?: HmiProperty<boolean>;
  headerText?: HmiMultilingualText;
  decimalPlaces?: HmiProperty<number>;
  widthMode?: HmiProperty<HmiDataGridColumnWidthMode>;
  sourceWidthMode?: string;
  width?: HmiProperty<number>;
  order?: HmiProperty<number>;
  filterDefinition?: HmiProperty<string>;
  sortDirection?: HmiProperty<HmiDataGridSortDirection>;
  sourceSortDirection?: string;
  sortPriority?: HmiProperty<number>;
}

File: src/BaseHmiTypes/Screens/Controls/HmiDataGridColumn.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiDataGridColumn
{
    public string? Name { get; set; }

    public string? DataType { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiMultilingualText? HeaderText { get; set; }

    public HmiProperty<int>? DecimalPlaces { get; set; }

    public HmiProperty<HmiDataGridColumnWidthMode>? WidthMode { get; set; }

    public string? SourceWidthMode { get; set; }

    public HmiProperty<double>? Width { get; set; }

    public HmiProperty<int>? Order { get; set; }

    public HmiProperty<string>? FilterDefinition { get; set; }

    public HmiProperty<HmiDataGridSortDirection>? SortDirection { get; set; }

    public string? SourceSortDirection { get; set; }

    public HmiProperty<int>? SortPriority { get; set; }
}
